src/MainMenu.cpp

```diff
@@ -1633,4 +1636,85 @@ ChainCallbackResult MainMenu::OnDraw(MainMenu *menu)
 }
 #pragma optimize("", on)
 
+#pragma optimize("s", on)
+CursorMovement MainMenu::MoveCursor(MainMenu *menu, i32 menuLenght)
+{
+    if (WAS_PRESSED_WEIRD(TH_BUTTON_UP))
+    {
+        menu->cursor--;
+        g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+        if (menu->cursor < 0)
+        {
+            menu->cursor = menuLenght - 1;
+        }
+        if (menu->cursor >= menuLenght)
+        {
+            menu->cursor = 0;
+        }
+        return CURSOR_MOVE_UP;
+    }
+
+    if (WAS_PRESSED_WEIRD(TH_BUTTON_DOWN))
+    {
+        menu->cursor++;
+        g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+        if (menu->cursor < 0)
+        {
+            menu->cursor = menuLenght - 1;
+        }
+        if (menu->cursor >= menuLenght)
+        {
+            menu->cursor = 0;
+        }
+        return CURSOR_MOVE_DOWN;
+    }
+
+    return CURSOR_DONT_MOVE;
+}
+#pragma optimize("", on)
+
+#pragma optimize("s", on)
+#pragma var_order(vm, d3dVec)
+ZunBool MainMenu::WeirdSecondInputCheck()
+{
+    i32 vm;
+    D3DXVECTOR3 d3dVec;
+
+    if (this->stateTimer < 0x1e)
+    {
+        return true;
+    }
+
+    if (!WAS_PRESSED_WEIRD(TH_BUTTON_SELECTMENU | TH_BUTTON_BOMB | TH_BUTTON_MENU | TH_BUTTON_Q | TH_BUTTON_S))
+    {
+        return true;
+    }
+
+    this->stateTimer = 0;
+    this->gameState = STATE_MAIN_MENU;
+    for (vm = 0; vm < 122; vm++)
+    {
+        this->vm[vm].pendingInterrupt = 2;
+    }
+    if (!(g_Supervisor.cfg.opts & 1 >> GCOS_USE_D3D_HW_TEXTURE_BLENDING))
```

Comment:
```suggestion
    if (!((g_Supervisor.cfg.opts >> GCOS_USE_D3D_HW_TEXTURE_BLENDING) & 1))
```

Reply:
I need to experiment with rewriting those as a bitfield. I bet that's what Zun did.